EncoreTickets.SDK/Utilities/DataStructures/Tree/ReadOnlyTree.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using EncoreTickets.SDK.Utilities.BaseTypesExtensions;
+
+namespace EncoreTickets.SDK.Utilities.DataStructures.Tree
+{
+    /// <summary>
+    /// The implementation of the ITree&lt;TKey, TValue&gt; interface.
+    /// It defines a tree that can be built based on the specified criteria and traversed afterwards.
+    /// The tree nodes can have an arbitrary number of children.
+    /// </summary>
+    /// <typeparam name="TKey"></typeparam>
+    /// <typeparam name="TValue"></typeparam>
+    public class ReadOnlyTree<TKey, TValue> : ITree<TKey, TValue>
```

Comment:
Are you sure this tree should be defined in the SDK? For me, it looks like a very specific tree that should have keys, key functions, etc. And it cannot be used for all types of possible trees. It might be better to identify it near the place where it is used.

Reply:
We have already defined stuff like specific cache implementations which are still not used anywhere. I don't see how this is less generic or potentially useful than that. Also why do you think it is very specific? It literally has nothing publicly visible apart from its factory methods and ITree impementation